find/finder.go

```diff
@@ -312,6 +318,7 @@ func (f *Finder) DatastoreList(ctx context.Context, path string) ([]*object.Data
 		if ref.Type == "Datastore" {
 			ds := object.NewDatastore(f.client, ref)
 			ds.InventoryPath = e.Path
+			ds.DatacenterPath = f.dc.InventoryPath
```

Comment:
@dnaeon in that case you don't need Finder, as SearchIndex.FindByInventoryPath does the job with absolute paths.  But I'd be fine with adding the nil check for now.  Are you using any of the Datastore upload/download methods?  Those need the Datacenter ref against VC, unless you set GOVMOMI_USE_SERVICE_TICKET=true